1.Pick the odd one out.

a)Brand datatype is object

b)Selling Price datatype is int64

c)Rating Price datatype is float64

d)Memory datatype is int

In [32]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3114 entries, 0 to 3113
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Brand           3114 non-null   object 
 1   Model           3114 non-null   object 
 2   Color           3114 non-null   object 
 3   Memory          3071 non-null   object 
 4   Storage         3075 non-null   object 
 5   Rating          2970 non-null   float64
 6   Selling Price   3114 non-null   int64  
 7   Original Price  3114 non-null   int64  
dtypes: float64(1), int64(2), object(5)
memory usage: 194.8+ KB
None


2.What is the most frequented Rating all Mobiles have.

a)4.5

b)4.1

c)4.4

d)4.3

In [1]:
import pandas as pd
df = pd.read_csv('/content/Mobiles.csv')
most_frequent_rating = df['Rating'].mode()[0]
print(most_frequent_rating)

4.3


3.What is the maximum Selling Price in the dataset.

a)180000

b)179900

c)179999

d)119999

In [2]:
maximum_selling_price = df['Selling Price'].max()
print(maximum_selling_price)

179900


4.What is the difference between Average of  Selling Price and  Original Price

a)1896.8474630700075

b)1876.8474630700075

c)1996.8474630700075

d)1897.8474630700075

In [4]:
average_selling_price = df['Selling Price'].mean()
average_original_price = df['Original Price'].mean()
difference = average_original_price - average_selling_price
print(difference)

1896.8474630700075


5.Which Mobiles has Highest frequency in the given dataset.

a)Apple

b)Samsung

c)Oppo

d)Redmi

In [6]:
highest_frequency = df['Brand'].mode()[0]
print(highest_frequency)

SAMSUNG


6.How many NaN values are there in Storage feature.

a)34

b)39

c)36

d)38

In [14]:
null_values = df['Storage'].isnull().sum()
print(null_values)

39


7.What is the difference between Max Selling Price and  Min Selling Price

a)179900

b)178900

c)178910

d)177900

In [16]:
max_selling_price = df['Selling Price'].max()
min_selling_price = df['Selling Price'].min()
difference = max_selling_price - min_selling_price
print(difference)

178900


8.What is the Cardinality (no. of unique values) of Color feature.

a)655

b)656

c)639

d)634

In [25]:
cardinality = df['Color'].nunique()
print(cardinality)

639


9.Apply .describe function on categorical columns and pick the odd one's out.

a)Top color is Balck

b)unique values in Memory is 27

c)frequency of Storage is 774

d)unique values in Brand is 18

In [27]:
print(df.describe(include=object))

          Brand       Model  Color Memory Storage
count      3114        3114   3114   3071    3075
unique       17         914    639     26      26
top     SAMSUNG  iPhone 11   Black   4 GB   64 GB
freq        719          36    489    750     774


10.What are the information's given in  .info function.

a)Memory usage of data frame

b)Missing values

c)Data type

d)Column names

In [28]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3114 entries, 0 to 3113
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Brand           3114 non-null   object 
 1   Model           3114 non-null   object 
 2   Color           3114 non-null   object 
 3   Memory          3071 non-null   object 
 4   Storage         3075 non-null   object 
 5   Rating          2970 non-null   float64
 6   Selling Price   3114 non-null   int64  
 7   Original Price  3114 non-null   int64  
dtypes: float64(1), int64(2), object(5)
memory usage: 194.8+ KB
None


11.How many Rows and Columns are present in the dataset.

a)3113 rows and 8 columns

b)3112 rows and 8 columns

c)3114 rows and 7 columns

d)3114 rows and 8 columns

In [30]:
print(df.shape)

(3114, 8)


12.What is the  percentage of 128 GB Mobiles in the given dataset.

a)25.17073171

b)18.04878049

c)24.84552846

d)23.74573171

In [31]:
total_mobiles = len(df)
mobiles_128GB = len(df[df['Storage'] == '128 GB'])
percentage = (mobiles_128GB / total_mobiles) * 100
print(percentage)

24.53436095054592


13.How many Rows are present in the dataset after dropping the duplicate values.

a)3006

b)3001

c)3005

d)3007

In [35]:
df_no_duplicates = df.drop_duplicates()
print(len(df_no_duplicates))

3006


14.How many Rows and Columns are present in the dataset after dropping all NaN values.

a)2865 rows × 7 columns

b)2897 rows × 8 columns

c)2897 rows × 7 columns

d)2987 rows × 8 columns

In [38]:
df_no_na = df.dropna()
print(df_no_na.shape)

(2897, 8)


Which of the following Mobile Brands are not in the given dataset.

a)Motorola

b)Lenovo

c)Nothing

d)Sony

In [41]:
existing_brand = df['Brand'].unique()
print(existing_brand)

['OPPO' 'HTC' 'IQOO' 'Google Pixel' 'LG' 'ASUS' 'realme' 'GIONEE' 'Nokia'
 'Apple' 'SAMSUNG' 'Lenovo' 'Motorola' 'POCO' 'vivo' 'Xiaomi' 'Infinix']
